### PR TITLE
Add GCSCopy task

### PR DESCRIPTION
**Pull request by jlowin**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?

Adds a task for copying files from one GCS location to another, without downloading them in between

### CODE REVIEW

File: src/prefect/tasks/google/storage.py [hunk 1]
```diff
@@ -256,3 +256,113 @@ def run(
         gcs_blob = self._get_blob(bucket, blob, encryption_key_secret)
         gcs_blob.upload_from_string(data)
         return gcs_blob.name
+
+
+class GCSCopy(Task):
+    """
+    Task template for copying data from one Google Cloud Storage blob to another, without
```
Comment (joshmeek) [inline src/prefect/tasks/google/storage.py]:
```suggestion
    Task template for copying data from one Google Cloud Storage bucket to another, without
```

File: src/prefect/tasks/google/storage.py [hunk 2]
```diff
@@ -256,3 +256,113 @@ def run(
...
+        - dest_blob (str, optional): default destination blob name; can be
+            overwritten at runtime.  Required prior to running the task.
+        - project (str, optional): default Google Cloud project to work within; can be overwritten at runtime.
+            If not provided, will be inferred from your Google Cloud credentials
+        - credentials_secret (str, optional): the name of the Prefect Secret
+            which stores a JSON representation of your Google Cloud credentials; can be overwritten at runtime.
+            Defaults to `GOOGLE_APPLICATION_CREDENTIALS`.
+        - **kwargs (dict, optional): additional keyword arguments to pass to the Task constructor
+
+    Note that the design of this task allows you to initialize a _template_ with default settings.  Each inidividual
```
Comment (joshmeek) [inline src/prefect/tasks/google/storage.py]: Not that it's needed here, and feel free to mark as resolved once you see this, but a good example of this either in the examples folder or right here in this docstring is a flow where a blob is copied from A -> B and B -> C; having the kwargs overwritten at runtime.

File: src/prefect/tasks/google/storage.py (hunk 1 above)
Comment (joshmeek) [reply src/prefect/tasks/google/storage.py]: _technically_ blob is correct but bucket feels correct to me. While it is duplicating a blob it's really copying a blob from one bucket to another.

File: src/prefect/tasks/google/storage.py (hunk 2 above)
Comment (jlowin) [reply src/prefect/tasks/google/storage.py]: 👍 added example

**Pull request merged**